src/core/Core.js

```diff
@@ -934,7 +942,15 @@ class Uppy {
    *
    * @return {Promise}
    */
-  upload (forceUpload) {
+  upload () {
+    if (!this.plugins.uploader) {
+      this.log('No uploader type plugins are used', 'warning')
+    }
+
+    if (Object.keys(this.state.files).length === 0) {
+      this.log('No files have been added', 'warning')
+    }
```

Comment:
We could do this for now, but this is not always a mistake (eg. transloadit plugin w/ an assembly that imports from somewhere).